Packages/com.unity.inputsystem/InputSystem/Unity.InputSystem.asmdef

```diff
@@ -55,6 +55,11 @@
             "name": "com.unity.ugui",
             "expression": "1.0.0",
             "define": "UNITY_INPUT_SYSTEM_ENABLE_UI"
+        },
+        {
+            "name": "com.unity.ugui",
+            "expression": "1.1.0",
+            "define": "UNITY_INPUT_SYSTEM_ENABLE_MULTIDISPLAY"
```

Comment:
@3iD0loN can you comment on this?  I don't have the answers.

It might make sense to rename the variable just so that it's clearer what it means though since, you're right, even if you do have 1.1.0, if you're on XBOX, for instance, it won't work.

Reply:
Yes, you are correct in that assumption.

UGUI has support for multiple displays right now, aside from touch the controls that we have added. As for UI Toolkit, I have done some preliminary tests and UI Toolkit documents are drawn on all screens regardless of settings, so there is no proper support for that in UI Toolkit.

Reply:
@jamesmcgill do you have any additional thoughts on this one?

Reply:
I'm also a little uneasy about adding platform defines around the code.
I'm still not 100% on their use so I don't have any ideas on how to remove/reduce them just now, but I would love it if we can look into ways to do that.

I'd hope that these defines are a temporary stop-gap?  And could be removed once we can rely on the required UGUI version.
But with the platform defines, I'm not clear on when we could sunset these. It feels like something should be pushed as standard into the platform and that the package should be using editor version defines until versions catch up. I'm not sure our package should be dealing with likely long-term platform inconsistencies.

Reply:
So we ended up only able to make this change to UGUI trunk so Unity 2023 onward will have the displayIndex set.  The reason we need these defines is that InputSystem supports all the way back to 2019 where we can't backport our UGUI changes.

Once InputSystem only supports 2023 and up, you could sunset these #defines.  Otherwise, I don't really see a way around them.

Reply:
Would we be able to remove the full define after 2023?
Also the `(UNITY_EDITOR_WIN || UNITY_EDITOR_OSX || UNITY_EDITOR_LINUX || UNITY_ANDROID)` part?

Not sure about this suggestion: but would a UNITY_VERSION >= 2023 define work here instead then?

Reply:
Yep, if InputSystem only supported 2023 onwards, we could remove the define. Also, if we backported this functionality we could remove the define and we'd like to do that if we can get permission.

This new feature only works in Editor and in Android.  I have to assume we'd like to see the functionality added to the other platforms at some point too, but not this time around.  Once the change is in the other platforms, we can get rid of the parenthetical condition you highlighted.

Reply:
I reached out to UGUI team to conclude whether downgrading UGUI package is supported or not. If its not we can stick to current defines without platform specifics. @EricFulton based on the modifications we made we would need to update Input System if changes are made so this feature is supported in older Unity versions, so please include us if such backports are made.

We can still get rid of the platform specifics for Input System (we have added these changes) since from Input System perspective the displayIndex will be zero if platform doesn't provide any value, e.g. UGUI have this field and its not platform specific.

From our perspective feel free to resolve this conversation.